components/director/internal/domain/label/repository.go

```diff
@@ -170,6 +170,30 @@ func (r *repository) DeleteAll(ctx context.Context, tenant string, objectType mo
 	return errors.Wrapf(err, "while deleting all Label entities from database for %s %s", objectType, objectID)
 }
 
+func (r *repository) DeleteByKey(ctx context.Context, tenant string, key string) error {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return errors.Wrap(err, "while fetching persistence from context")
+	}
+
+	stmt := fmt.Sprintf(`DELETE FROM %s WHERE "key" = $1 AND "tenant_id" = $2`, tableName)
+	result, err := persist.Exec(stmt, key, tenant)
+	if err != nil {
+		return errors.Wrapf(err, `while deleting all Label entities from database with key "%s"`, key)
+	}
+
+	rows, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "while receiving rows affected by query")
+	}
+
+	if rows == 0 {
```

Comment:
What if someone tries deleting LD with "force flag" and there's no labels with that key? Should that result in error?

Reply:
changed it, it won't result with an error